pkg/auth/handler/ssohandler/iframe_handler.go

```diff
@@ -0,0 +1,52 @@
+package ssohandler
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/skygeario/skygear-server/pkg/auth"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/sso"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz/policy"
+	"github.com/skygeario/skygear-server/pkg/core/inject"
+	"github.com/skygeario/skygear-server/pkg/core/server"
+)
+
+func AttachIFrameHandlerFactory(
+	server *server.Server,
+	authDependency auth.DependencyMap,
+) *server.Server {
+	server.Handle("/sso/iframe_handler", &IFrameHandlerFactory{
+		authDependency,
+	}).Methods("GET")
```

Comment:
Since the endpoint is the iframe source but not an `XMLHttpRequest`, so pre-flight request should not be needed. 

In the login flow, this endpoint has js script to call `/sso/config` api to send the result back to the parent window. Since in `iframe_handler`, we initialized the js sdk without api key and `/sso/config` should not be api key required.

Reply:
Updated and thanks @kaojohnny and @Steven-Chan. Feel free to let me know for any questions:)